src/main/java/oss/fosslight/controller/SearchController.java

```diff
@@ -0,0 +1,177 @@
+package oss.fosslight.controller;
+/*
+ * Copyright (c) 2021 Suram Kim
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.*;
+import oss.fosslight.CoTopComponent;
+import oss.fosslight.common.CoConstDef;
+import oss.fosslight.domain.LicenseMaster;
+import oss.fosslight.common.Url.SEARCH;
+import oss.fosslight.domain.OssMaster;
+import oss.fosslight.domain.PartnerMaster;
+import oss.fosslight.domain.Project;
+import oss.fosslight.service.SearchService;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.*;
+
+@Controller
+@Slf4j
+public class SearchController extends CoTopComponent {
+
+
+    @Autowired SearchService searchService;
+
+
+    @GetMapping(value=SEARCH.LICENSE)
+    public @ResponseBody ResponseEntity<Object> saveLicenseSearchFilter(LicenseMaster licenseMaster
+            , HttpServletRequest req
+            , HttpServletResponse res){
+
+        int page = Integer.parseInt(req.getParameter("page"));
+        int rows = Integer.parseInt(req.getParameter("rows"));
+        String sidx = req.getParameter("sidx");
+        String sord = req.getParameter("sord");
+
+        licenseMaster.setCurPage(page);
+        licenseMaster.setPageListSize(rows);
+        licenseMaster.setSortField(sidx);
+        licenseMaster.setSortOrder(sord);
+
+        searchService.saveLicenseSearchFilter(licenseMaster, loginUserName());
+
+        return makeJsonResponseHeader("saved");
+
+    }
+
+
+    @GetMapping(value=SEARCH.OSS)
+    public @ResponseBody ResponseEntity<Object> saveOssSearchFilter(OssMaster ossMaster,
+                                                                    HttpServletRequest req){
+
+        int page = Integer.parseInt(req.getParameter("page"));
+        int rows = Integer.parseInt(req.getParameter("rows"));
+        String sidx = req.getParameter("sidx");
+
+        if(sidx != null) {
+            sidx  = sidx.split("[,]")[1].trim();
+        }
+
+        ossMaster.setSidx(sidx);
+        ossMaster.setCurPage(page);
+        ossMaster.setPageListSize(rows);
+
+        List<String> values = Arrays.asList("https://", "http://", "www.");
```

Comment:
It will be better to set specific name to values such as httpPrefixes. And it is good way to make method to retrieve homepage from 70 to 73 line.